for_developers/product_design.md

```diff
@@ -0,0 +1,114 @@
+# Product Design
+
+## Our Product Mission
+
+1. **Provide a deep learning model training framework that encompasses various tasks solvable by deep learning models.** Our product aims to be convenient for developers, allowing them to easily expand support for tasks and models.
+
+2. **Empower users to seamlessly and intuitively navigate the OpenVINO™-based workflow, from model training to inference.** To ensure a user-friendly experience throughout, our product should have user-friendly CLI and Python API entrypoints.
+
+3. **Support Intel devices for accelerating deep learning model training:** We offer best practices for training from scratch or fine-tuning popular model architectures using Intel devices.
+
+## Product Design Principles
+
+To realize our product missions, we establish the following design principles and details, providing a comprehensive overview of our product's structure.
+
+### Core
+
+1. **Static Typing System for Task-Data-Model to Accelerate Task and Model Development Cycles**
+
+   _"Advocates of static typing argue that ... a better design time developer experience..."_[^1]
+
+   Our framework should support various deep learning tasks simultaneously.
+   To achieve this mission, it is essential to make it developer-friendly for expanding our product in both task and model dimensions.
+   We believe that nice abstractions with static typing can help developers in this regard.
+   We introduced the **Task-Data-Model** abstraction with static typing.
```

Comment:
How about `Therefore, we here introduce the **Task-Data-Model** ...`?

Reply:
```suggestion
   Therefore, we here introduce the **Task-Data-Model** abstraction with static typing.
```